src/js/compat.ts

```diff
@@ -31,48 +34,61 @@ export async function initNodeModules() {
     return;
   }
   // @ts-ignore
-  nodeUrlMod = (await import("url")).default;
-  nodeFsPromisesMod = await import("fs/promises");
+  nodeUrlMod = (await import("node:url")).default;
+  // @ts-ignore
+  nodeFsPromisesMod = await import("node:fs/promises");
   if (globalThis.fetch) {
     nodeFetch = fetch;
   } else {
     // @ts-ignore
-    nodeFetch = (await import("node-fetch")).default;
+    nodeFetch = (await import("node:node-fetch")).default;
   }
   // @ts-ignore
-  nodeVmMod = (await import("vm")).default;
-  nodePath = await import("path");
+  nodeVmMod = (await import("node:vm")).default;
+  nodePath = await import("node:path");
   pathSep = nodePath.sep;
 
   // Emscripten uses `require`, so if it's missing (because we were imported as
   // an ES6 module) we need to polyfill `require` with `import`. `import` is
   // async and `require` is synchronous, so we import all packages that might be
   // required up front and define require to look them up in this table.
-
-  if (typeof require !== "undefined") {
+  //
+  // When importing Deno as a module, require exists here but it does not exist
+  // in pyodide.asm.js, not sure why. Anyways in that case go ahead and define
+  // it.
+  if (typeof require !== "undefined" && !IN_DENO) {
     return;
   }
+
   // These are all the packages required in pyodide.asm.js. You can get this
   // list with:
   // $ grep -o 'require("[a-z]*")' pyodide.asm.js  | sort -u
-  const fs = await import("fs");
-  const crypto = await import("crypto");
-  const ws = await import("ws");
-  const child_process = await import("child_process");
+  const fs = await import("node:fs");
+  const crypto = await import("node:crypto");
+  // @ts-ignore
+  const tty = await import("node:tty");
+  // Stub these out in deno because otherwise I get:
+  // TypeError: Loading unprepared module: npm:ws
+  const ws = IN_DENO ? {} : await import("ws");
+  const child_process = IN_DENO ? {} : await import("child_process");
```

Comment:
Interesting. Does it mean we are not using these two modules actually?

Reply:
Deno tries to be more like a browser context. It has globals for the WHATWG apis like WebSocket, URL, and fetch already in the globalThis. Shouldn't need the npm packages in those cases

Reply:
Well they are pretty seldomly used. `ws` is used by `-lwebsocket.js`. `child_process` is used by `system` / `execl`. It should be possible to import them in deno, but I am having trouble getting them to work, and working minus some rare features is probably better than not working at all.

Reply:
> Shouldn't need the npm packages / node imports in those cases

Well sure, but only if we are willing to write deno-specific code. In the short term, it would be nice to go through the node polyfills unless there is a pressing reason not to. Longer term maybe we can add deno-specific code, but as long as we are only building one version of Pyodide any deno-specific code also has to be downloaded but not used by browsers so there are reasons not to duplicate if there is a way to make it work.